### DISCUSSION TITLE
Query parameter serialisation

### DISCUSSION THREAD

**Discussion by Strydom** [vikejs/vike#1295] [Discussion]
Hello,
I'm not sure if I'm doing something wrong or if there is something in the flow that is incorrectly serialising the URL's.

I'm updating the query params to perform a search, one of them is equal to `filter=Coral reefs`
```
const search = () => {
  const newSearch = `?${new URLSearchParams(array)}`
  console.log('newSearch', newSearch)
  if (window.location.search != newSearch) {
    window.location.replace(`${newSearch}`)
  }
}
```

The search function changes this to `filter=Coral+reefs`

When i receive the query in the `onBeforeRender` it has encoded the `+` as `%2B`.
This is then being decoded to a `+` again.

Afaik it should not be encoded as `%2B`? It should remain as `+` which would then be decoded to a space.

I'm running with `vercel dev`

**Comment by brillout** [comment]
Are you sure it's related to Vike? Minimal reproduction welcome.
